proseco/preview.py

```diff
@@ -0,0 +1,317 @@
+        self.stars = stars
+
+        fig = plt.figure(figsize=(4.5, 4))
+        ax = fig.add_subplot(1, 1, 1)
+        self.acqs.stars
+        self.plot(ax=ax)
+        plt.tight_layout()
+        fig.savefig(str(outfile))
+        plt.close(fig)
+
+    def get_text_pre(self):
+        """Get pre-formatted text for report."""
+
+        P2 = -np.log10(self.acqs.calc_p_safe())
+        att = Quat(self.att)
+        self._base_repr_()
+        catalog = '\n'.join(self.pformat(max_width=-1))
+        self.acq_count = np.sum(self.acqs['p_acq'])
+        self.guide_count = self.guides.guide_count(self.guides['mag'], self.guides.t_ccd)
+
+        message_text = self.get_formatted_messages()
+
+        text_pre = f"""\
+{self.detector} SIM-Z offset: {self.sim_offset}
+RA, Dec, Roll (deg): {att.ra:.6f} {att.dec:.5f} {att.roll:.5f}
+Dither acq: Y_amp= {self.dither_acq.y:.1f}  Z_amp={self.dither_acq.z:.1f}
+Dither gui: Y_amp= {self.dither_guide.y:.1f}  Z_amp={self.dither_guide.z:.1f}
+Maneuver Angle: {self.man_angle:.2f}
+Date: {self.date}
+
+{catalog}
+
+{message_text}\
+Probability of acquiring 2 or fewer stars (10^-x): {P2:.2f}
+Acquisition Stars Expected: {self.acq_count:.2f}
+Guide Stars count: {self.guide_count:.2f}
+Predicted Guide CCD temperature (max): {self.t_ccd_guide:.1f}
+Predicted Acq CCD temperature (init) : {self.t_ccd_acq:.1f}"""
+
+        return text_pre
+
+    def get_formatted_messages(self):
+        """Format message dicts into pre-formatted lines for the preview report"""
+
+        lines = []
+        for message in self.messages:
+            category = message['category']
+            idx_str = f"[{message['idx']}] " if ('idx' in message) else ''
+            line = f">> {category.upper()}: {idx_str}{message['text']}"
+            line = stylize(line, category)
+            lines.append(line)
+
+        out = '\n'.join(lines) + '\n\n' if lines else ''
+        return out
+
+    def add_row_col(self):
+        """Add row and col columns if not present"""
+        if 'row' in self.colnames:
+            return
+
+        row, col = yagzag_to_pixels(self['yang'], self['zang'], allow_bad=True)
+        index = self.colnames.index('zang') + 1
+        self.add_column(Column(row, name='row'), index=index)
+        self.add_column(Column(col, name='col'), index=index + 1)
+
+    def preview(self):
+        """Prelim review of ``self`` catalog.
+
+        This is based on proseco and (possibly) other available products, e.g. the DOT.
+        """
+        self.make_starcat_plot()
+        self.add_row_col()
+        self.check_catalog()
+        self.context['text_pre'] = self.get_text_pre()
+
+    def check_catalog(self):
+        for entry in self:
+            self.check_position_on_ccd(entry)
+
+        self.check_guide_geometry()
+        self.check_acq_p2()
+
+    def check_guide_geometry(self):
+        """Check for guide stars too tightly clustered
+
+        (1) Check for any set of n_guide-2 stars within 500" of each other.
```

Comment:
The n-2 check seems a fine place to start.  Not sure how to figure out a reasonable tolerance but 500" doesn't seem too crazy.